src/org/skypro/skyshop/basket/ProductBasket.java

```diff
@@ -2,23 +2,23 @@
 
 import org.skypro.skyshop.product.Product;
 
-import java.util.Iterator;
-import java.util.LinkedList;
-import java.util.List;
+import java.util.*;
 
 public class ProductBasket {
-    private List<Product> products = new LinkedList<>();
+    private Map<String, ArrayList<Product>> products = new HashMap<>();
 
     public void addProduct(Product product) {
-        if (product != null) {
-            products.add(product);
+        if (product == null) return;
+        if (!products.containsKey(product.getName())) {
+            products.put(product.getName(), new ArrayList<>());
         }
+        products.get(product.getName()).add(product);
     }
 
     public int basketCost() {
         int sum = 0;
-        for (Product product : products) {
-            if (product != null) {
+        for (ArrayList<Product> productList : products.values()) {
```

Comment:
```suggestion
        for (List<Product> productList : products.values()) {
```
не забываем про полиморфизм